harvesters/src/main/java/org/fao/geonet/kernel/harvest/harvester/fragment/FragmentHarvester.java

```diff
@@ -351,19 +362,25 @@ private void createOrUpdateSubtemplate(Element fragment) throws Exception {
 
         String schema = fragment.getAttributeValue("schema");
 
+        String title = fragment.getAttributeValue("title");
```

Comment:
Hi Francois, 

Could use a strategy like that used in index-subtemplate.xsl to come up with a title yes but for clients that load variations of the same subtemplate type (eg. contacts for resources versus contacts for author in citation) they can specify the title and uuid using this method so that filters for the directoryentry selector can be used to subset the subtemplates as required. 

For example, they can specify a filter on a directoryentry selector for a contact in the metadata citation to look for author contacts and a different filter for a directoryentry selector in the data resource contact. There is no easy way to distinguish between these contacts using the index-subtemplate.xsl approach: they are all contacts.

I do agree though that we should take the title field in the db and index it with the subtemplates as lucene field _title to avoid inconsistency between the db and the lucene index. That way both could be used in that if we have a title in the db then that is used otherwise we generate one using the rules in in index-subtemplate.xsl.